### DISCUSSION TITLE
Mission 8: Save station invisible until powered up transit

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#147] [General]
I moved the save station in mission 8 upper transit room to be next to the elevator.
After figuring there is a 35 point difference in the X coordinate and correcting for it I managed to spawn 
a working save point, but the housing was invisible. It spawned the housing of the save station after I used 
the computer that opens the door for the android.

![image](https://user-images.githubusercontent.com/51859157/220753176-176f321a-44a1-4d80-9fe2-069a62de5bbc.png)
![image](https://user-images.githubusercontent.com/51859157/220753547-c54df124-a5da-4a68-9819-4983860f4a97.png)

I made a short video clip after saving at the invisible save station.  
https://youtu.be/O-MmwuPM96U

It's curious what the power on is triggering that also makes the save point housing load in too ?

**Comment by ghramsey** [comment]
This get stranger and stranger.  There must be code that loads something in Transit Control.  Maybe it's a zone thing but if 
so it's not appearant where the boundary is or how to identify it.

Loading M8 in debug mode also has the same issue. The save point fails to load the housing but is otherwise usable.
I had to clip through the door to get to the lift upstairs but the other save station in transit and by medical entry are
present; this one is not.

**Comment by ghramsey** [comment]
I'm at a loss as to why merely repositioning the existing save point caused so much issue.
I created a new save point and after some fine tuning with the 35 point offset to get the location right works OK now.

**Comment by MattFiler** [comment]
This definitely sounds like a zoning issue. Perhaps I should add in links to TriggerSequence and CAGEAnimation entities in the new "references" tab to be able to easily spot which zones entities are linked to?

For now, I'd recommend just adding a new save station rather than moving one, as you can then have it spawned all the time rather than zoned.

**Comment by ghramsey** [comment]
I ended up doing just that; Adding a new save station vs moving the existing one whch I can just set to be off screen 
with it's coordinates at 255,255,255 if I want.   I'm at a loss about the zoning works on this map.   
I understand it's a division of the area, but in most other maps it's clearly defined in the Levels composite.
For SCI_Hub there's only one entry at the bottom of the levels heirchy and that is Hub_section.   The thing I can ID as 'zoning' 
seems to be in that there is a 'map_keyframe' parameter on each of the two stock save points which is a string that is SCI_Hub_1
 for the upper transit level and SCI_Hub_2 for the transit platform level.  I cannot find how those work out into zones. 
 They don't appear in that format anywhere in the environment entriess.  And the save station I added by the elevator leading 
to the hospital map did not require I add a map keyframe parameter to both appear and function.

Looking at the SAVE_TERMINAL map keyframe defined as a string variable and leads out to mapitem.
So that seems to be relevent to this issue.   The thing is the map keyframe on the same level should be the same right?
It would not change just from moving it's position marker. If I moved it downstairs I could understand it.
In any case this is solved in as much as I did create a new savepoint.  Just a curiosity.

**Reply by MattFiler** [reply]
`map_keyframe` is the parameter for setting the map GFX file in the tab menu - not related to anything to do with zones.

The hook you saw for it the save station would be to add the save station icon to the map.

**Reply by ghramsey** [reply]
OK. That's kinda what was thinking it did.  So where and how does this map setup zones? 

Upper and Lower sections perhaps as I can see those being a boundary.  There is definitely a transition since the game 
fades out during the elevator but that might be window just effects; The traverse in the vent is only a short trip down.

For kicks, I tried to spawn the save point in the environment vs in the hub_section just for kicks either that composite doesn't 
have the same X offset and I was therefore placing it wrong or it just won't spawn from there.  It had to be placed through hub_section.

**Reply by MattFiler** [reply]
I've just pushed an update to the staging branch which will now allow you to see TriggerSequence entities that reference the currently selected entity. This should be super useful for figuring out zones!

**Reply by ghramsey** [reply]
That's interesting. Thank you.  It might also help cipher out the LV426_PT2 thing which still has me scratching my head and yanking out clumps of hair over it.
